blockchain/blockchain.go

```diff
@@ -143,6 +143,18 @@ func head(txn db.Transaction) (*core.Block, error) {
 	return BlockByNumber(txn, height)
 }
 
+func headsHeader(txn db.Transaction) (*core.Header, error) {
+	height, err := chainHeight(txn)
+	if err != nil {
+		return nil, err
+	}
+	block, err := BlockByNumber(txn, height)
```

Comment:
```suggestion
	header, err := blockHeaderByNumber(txn, height)
```

Reply:
You should use the `headsHeader` in `func (b *Blockchain) HeadsHeader()` as well. Since now they are implementing the same logic.

Reply:
Hey there, I added the changes you suggested in the latest commit. Please let me know if there's a mistake.